js/src/data/api-fetch-middlewares.test.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Internal dependencies
+ */
+import { createErrorResponseCatcher } from './api-fetch-middlewares';
+import { API_NAMESPACE } from './constants';
+
+describe( 'createErrorResponseCatcher', () => {
+	// Ref: https://github.com/WordPress/gutenberg/blob/%40wordpress/api-fetch%405.1.1/packages/api-fetch/src/index.js#L68-L81
+	const checkStatus = ( response ) => {
+		if ( response.status >= 200 && response.status < 300 ) {
+			return response;
+		}
+
+		throw response;
+	};
+
+	const createFetchHandler = ( status, jsonParsedData ) => () => {
+		const response = {
+			status,
+			json() {
+				return Promise.resolve( jsonParsedData );
+			},
+		};
+		return Promise.resolve( response ).then( checkStatus );
+	};
+
+	const passReject = ( response ) => Promise.reject( response );
+
+	const optionsDefaultParse = { path: `${ API_NAMESPACE }/hi` };
+	const optionsDontParse = { path: `${ API_NAMESPACE }/hi`, parse: false };
+
+	describe( 'force overwrite `parse` option to `false` and return the same result', () => {
+		let middleware;
+
+		beforeEach( () => {
+			middleware = createErrorResponseCatcher( passReject );
+		} );
+
+		it( 'should only handle requests where the path belongs to this plugin', async () => {
```

Comment:
:question:
"Handle" is quite ambiguous. After looking at the implementation I'd expect that for other namespaces the callback should not be called at all. But here we are expecting it to be called

Reply:
I found the test case itself probably too ambiguous to know the test intention, so I made a change https://github.com/woocommerce/google-listings-and-ads/pull/834/commits/fbdc193f9b4c0370bd5b98765df6e5020dce92a0 to see if it more clear